### PR TITLE
Fixes deadletter related tickets

### PR DESCRIPTION
**Pull request by tapasweni-pathak**
Things related to deadletter, cli.

Closes https://github.com/mapbox/ecs-watchbot/issues/141.
Closes https://github.com/mapbox/ecs-watchbot/issues/159.
Closes https://github.com/mapbox/ecs-watchbot/issues/192.
Closes https://github.com/mapbox/ecs-watchbot/issues/193.

- [ ] Add Tests.

cc @mapbox/platform-engine-room.

### CODE REVIEW

File: mappings/ecr-region.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    },
+    "ap-southeast-1": {
+        "Region": "us-west-2"
+    },
+    "ap-southeast-2": {
+        "Region": "us-west-2"
+    },
+    "us-east-2": {
+        "Region": "us-east-1"
+    }
```
[the remark this answers is not in the document]
Comment (tapasweni-pathak) [reply mappings/ecr-region.json]: Right, deleted the file.

[the remark this answers is not in the document]
Comment (tapasweni-pathak) [reply mappings/ecr-region.json]: Per my understanding ECR is available in all regions but ecs-conex images are stored in few regions, would it be needed to define the mapping while creating a new ecs-watchbot service in a region as to where ecs-conex is writing images?

[the remark this answers is not in the document]
Comment (jakepruitt) [reply mappings/ecr-region.json]: @tapasweni-pathak ooo good catch, I didn't check line 339 of lib/template before.

**Comment by tapasweni-pathak** [comment]
Update: To test this run:

`watchbot-dead-letter stackName regionName`, not a bin/cli.js command.



The first one has 5 options, all other have 1 option in the dropdown menu. Currently the test fails as `TypeError: reciever.on is not a function` functionality is failing, one of the option in 1st dropdown option. It is not a function, looks simple to debug and fix, debugging  :ok_woman:

```js
  function writeOut(sqs, queue) {                                                 
    const reciever = receiveAll(sqs, queue.deadLetter);                           
    const stringifier = new stream.Transform({                                    
      objectMode: true,                                                           
      transform: function(msg, _, callback) {                                     
        const data = { subject: msg.subject, message: msg.message };              
        this.push(`${JSON.stringify(data)}\n`);                                   
        stringifier.handles.push(msg.handle);                                     
        callback();                                                               
      }                                                                           
    });                                                                           
    stringifier.handles = [];                                                     
    return new Promise((resolve, reject) => {                                     
      const done = returnMany(sqs, queue.deadLetter, stringifier.handles)         
        .then(() => resolve())                                                    
        .catch((err) => reject(err));                                             
      reciever                                                                    
        .on('error', reject)                                                      
        .pipe(stringifier)                                                        
        .on('error', reject)                                                      
        .on('end', done)                                                          
        .pipe(process.stdout);                                                    
    });                                                                           
  }  
```

**Comment by tapasweni-pathak** [comment]
Update: To test this run:

`watchbot-dead-letter stackName regionName`, not a bin/cli.js command.



The first one has 5 options, all other have 1 option in the dropdown menu. Currently the test fails as the call to one of the dropdown option in first options fails with `TypeError: reciever.on is not a function`. Looks simple to debug and fix node8 thingy? debugging  :ok_woman:

```js
  function writeOut(sqs, queue) {                                                 
    const reciever = receiveAll(sqs, queue.deadLetter);                           
    const stringifier = new stream.Transform({                                    
      objectMode: true,                                                           
      transform: function(msg, _, callback) {                                     
        const data = { subject: msg.subject, message: msg.message };              
        this.push(`${JSON.stringify(data)}\n`);                                   
        stringifier.handles.push(msg.handle);                                     
        callback();                                                               
      }                                                                           
    });                                                                           
    stringifier.handles = [];                                                     
    return new Promise((resolve, reject) => {                                     
      const done = returnMany(sqs, queue.deadLetter, stringifier.handles)         
        .then(() => resolve())                                                    
        .catch((err) => reject(err));                                             
      reciever                                                                    
        .on('error', reject)                                                      
        .pipe(stringifier)                                                        
        .on('error', reject)                                                      
        .on('end', done)                                                          
        .pipe(process.stdout);                                                    
    });                                                                           
  }  
```

**Comment by tapasweni-pathak** [comment]
Update: To test this run:

`watchbot-dead-letter stackName regionName`, not a bin/cli.js command.



`Triage dead messages individually?` has 5 inquirer dropdown options, all other have 1 option in the dropdown menu. Currently the test fails as the call to one of the dropdown option in first options fails with `TypeError: reciever.on is not a function`. Looks simple to debug and fix node8 thingy? 
  debugging :ok_woman:

```js
  function writeOut(sqs, queue) {                                                 
    const reciever = receiveAll(sqs, queue.deadLetter);                           
    const stringifier = new stream.Transform({                                    
      objectMode: true,                                                           
      transform: function(msg, _, callback) {                                     
        const data = { subject: msg.subject, message: msg.message };              
        this.push(`${JSON.stringify(data)}\n`);                                   
        stringifier.handles.push(msg.handle);                                     
        callback();                                                               
      }                                                                           
    });                                                                           
    stringifier.handles = [];                                                     
    return new Promise((resolve, reject) => {                                     
      const done = returnMany(sqs, queue.deadLetter, stringifier.handles)         
        .then(() => resolve())                                                    
        .catch((err) => reject(err));                                             
      reciever                                                                    
        .on('error', reject)                                                      
        .pipe(stringifier)                                                        
        .on('error', reject)                                                      
        .on('end', done)                                                          
        .pipe(process.stdout);                                                    
    });                                                                           
  }  
```

**Comment by jakepruitt** [comment]
Alright, gave a shot at refactoring some of this, but there's definitely more room for improvement if you want to pick any of this up @arunasank!

**Comment by jakepruitt** [comment]
Alright, I think all of the tests are passing now. This still needs real-life tests against dead letters with messages in them before we can merge.

**Comment by arunasank** [comment]
👋 @jakepruitt Is there a reason you renamed the `test/*.jest.js` files to `test/*.spec.js` as part of this PR? We should update the test scripts in `package.json` to account for this change.

**Pull request merged**